### ISSUE TITLE
Nano : _check_data_type(y) can be removed or checked conditionally?

### ISSUE THREAD

**Issue by rnwang04** [intel-analytics/BigDL#4659]
If I just quantize my model and don't tune, maybe the value of y is not important?
In my case, the y of the dataset is a dict, but after I remove this line https://github.com/intel-analytics/BigDL/blob/f682c06c54f7bf4afffd4ce10789bee5564bc9c9/python/nano/src/bigdl/nano/deps/neural_compressor/pytorch/dataloader.py#L37
it works.

**Comment by TheaperDeng** [comment]
yeah, we should remove this check for those users who don't tune their quantization config

**Comment by zhentaocc** [comment]
The intension of this check is to make sure the error of feeding input and forward can be raised before execution. So maybe just remove `_check_data_type(y)` for now? And to enhance the checking, we can include metric to help validate the dataloader format.
